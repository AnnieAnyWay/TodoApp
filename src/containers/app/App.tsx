import React from "react";
import TodoList from "../../components/todo/todo-list.block";
import Header from "../../components/header/header.block";

function App() {
  return (
    <div>
      <Header />
      <TodoList />
    </div>
  );
}

export default App;
