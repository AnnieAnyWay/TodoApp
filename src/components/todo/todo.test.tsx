import React from "react";
import renderer from "react-test-renderer";
import { combineReducers, createStore, compose, applyMiddleware } from "redux";
import thunkMiddleware from "redux-thunk";
import rootReducer, { initialState } from "../../store/reducers";

import { Provider } from "react-redux";
import App from "../../containers/app/App";
import TodoList from "./todo-list.block";
import TodoListItem from "./todo-list-item.block";

function create(state = {}) {
  return createStore(
    rootReducer,
    {
      ...initialState,
      ...state,
    },
    compose(applyMiddleware(thunkMiddleware))
  );
}

const mockedStore = {
  todos: [
    {
      description: "Run NY marathon later",
      dueDate: "2021-03-27T13:30:00.000Z",
      id: "7",
      isComplete: false,
    },
    {
      description: "Run LA marathon",
      dueDate: "2021-03-30T13:30:00.000Z",
      id: "8",
      isComplete: false,
    },
  ],
  ui: { isLoading: false },
};

it("renders correctly when there are no items", () => {
  const store = create();
  const snap = renderer
    .create(
      <Provider store={store}>
        <TodoList />
      </Provider>
    )
    .toJSON();
  expect(snap).toMatchSnapshot();
});

it("renders correctly with items", () => {
  const store = create(mockedStore);
  const snap = renderer
    .create(
      <Provider store={store}>
        <TodoList />
      </Provider>
    )
    .toJSON();
  expect(snap).toMatchSnapshot();
});

it("renders correctly todo item", () => {
  const store = create(mockedStore);
  const snap = renderer
    .create(
      <Provider store={store}>
        <TodoListItem id={"7"} />
      </Provider>
    )
    .toJSON();
  expect(snap).toMatchSnapshot();
});
