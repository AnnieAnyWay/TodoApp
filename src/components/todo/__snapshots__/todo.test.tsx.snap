// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly todo item 1`] = `
<li
  className="todo-item  "
>
  <input
    checked={false}
    onChange={[Function]}
    type="checkbox"
  />
  <span
    className="todo-item__name"
  >
    Run NY marathon later
  </span>
  <span
    className="todo-item__due-date"
  >
    3/27/2021
  </span>
</li>
`;

exports[`renders correctly when there are no items 1`] = `null`;

exports[`renders correctly with items 1`] = `
Array [
  <ul
    className="todo-list"
  >
    <li
      className="todo-item  "
    >
      <input
        checked={false}
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="todo-item__name"
      >
        Run NY marathon later
      </span>
      <span
        className="todo-item__due-date"
      >
        3/27/2021
      </span>
    </li>
    <li
      className="todo-item  "
    >
      <input
        checked={false}
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="todo-item__name"
      >
        Run LA marathon
      </span>
      <span
        className="todo-item__due-date"
      >
        3/30/2021
      </span>
    </li>
  </ul>,
  <div
    className="loader-container"
  >
    <div
      className="loader"
    >
      <div />
      <div />
    </div>
  </div>,
]
`;
